astropy/config/paths.py

```diff
@@ -33,8 +33,11 @@ def _find_home():
         if 'HOME' in os.environ:
             homedir = os.environ['HOME']
         else:
-            raise OSError('Could not find unix home directory to search for '
-                          'astropy config dir')
+            try:
```

Comment:
Can this entire function be effectively replaced by `os.expanduser('~')`?  This claims to be applicable for all platforms so it seems best to delegate to this core library function.  I guess it isn't clear if all the custom logic below for Windows is reflected in that function.

Reply:
Looking at the code for `expanduser` it takes a very different approach and uses the `pwd` module (password database) to get the home directory if `HOME` is not defined.  That seems at first glance reliable, maybe?

Reply:
I think it could - I took the current approach to maintain any previous behaviour, but in ipython they just use `os.expanduser`. If there's openess to changing it I'm happy to modify in that direction?

Reply:
I recall that @embray had good Windows knowledge back in the day.  Git blame confirms that some of the code came from @embray.  Thoughts?

Reply:
I'm not exactly sure what you're asking me, but IIRC using [`os.expanduser`](https://docs.python.org/3.6/library/os.path.html#os.path.expanduser) on Windows is reasonable.

Reply:
👍 if there is consensus that `os.expanduser('~')` is a reasonable approach to take then I will change the PR and see if CI is happy.

Reply:
@embray - my question is whether we can replace the *entire* `_get_home` code with a single line `return os.expanduser('~')`.  Are all those special cases on Windows still necessary or are they a relic of how things were 10 years ago?

Reply:
Is there any chance this change can be merged (to fix the issues we are having in sunpy, whilst definitely maintaining back-compat), and discussion about whether to replace the whole function with `os.path.expanduser('~')` moved to a new issue?

Reply:
Yes, good plan, sorry for the delay. Can you open that issue? For merging this needs a bugfix changelog entry indicating fix of problems in MacOS with no HOME dir var.  I can't see any reason not to backport to LTS, so put it in the 4.0.2 section.